src/Node/ReadLine.purs

```diff
@@ -1,42 +1,86 @@
 -- | This module provides a binding to the Node `readline` API.
 
-module Node.ReadLine where
+module Node.ReadLine
+  ( Interface
+  , InterfaceOptions
+  , Completer
+  , LineHandler
+  , createInterface
+  , createInteractiveConsoleInterface
+  , noCompletion
+  , prompt
+  , setPrompt
+  , setLineHandler
+  , close
+  ) where
 
 import Prelude (return)
-
-import Control.Monad.Eff
-import Control.Monad.Eff.Console
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Node.Process (stdin, stdout)
+import Node.Stream (Readable, Writable)
 
 -- | A handle to a console interface.
 -- |
 -- | A handle can be created with the `createInterface` function.
 foreign import data Interface :: *
 
+foreign import createInterfaceImpl :: forall opts eff. { | opts } -> Eff ( | eff) Interface
+
+foreign import undefined :: forall a. a
+
+-- | Options passed to `readline`'s `createInterface`
+type InterfaceOptions r w a e =
+  { input :: Readable r ( | e)
+  , output :: Maybe (Writable w ( | e))
+  , completer :: Maybe (Completer a)
+  , terminal :: Maybe Boolean
+  , historySize :: Maybe Int
+  }
+
 -- | A function which performs tab completion.
 -- |
--- | This function takes the partial command as input, and returns a collection of 
+-- | This function takes the partial command as input, and returns a collection of
 -- | completions, as well as the matched portion of the input string.
-type Completer eff = String -> Eff (console :: CONSOLE | eff) { completions :: Array String, matched :: String }
+type Completer eff = String -> Eff ( | eff) { completions :: Array String, matched :: String }
 
--- | A function which handles input from the user.
-type LineHandler eff a = String -> Eff (console :: CONSOLE | eff) a 
+-- | Builds an interface with the specified options.
+createInterface :: forall r w a eff. InterfaceOptions r w a eff -> Eff ( | eff) Interface
```

Comment:
You may be right.  I made that split when I first started running into errors regarding effect rows that I didn't understand.  It made me think that it may not make sense to constrain the interface and callbacks to the same list of effects.  Does that make sense?  Like, if the completer needed to make an http call to get the completion data.